reactComponents/src/searchbar/tokenSearch/Chip.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { memo, useContext } from 'react';
+import styled from 'styled-components';
+import TokenSearchContext from '../Context/TokenSearch';
+
+
+const StyledChip = styled.div`
+  > input {
+    display: none;
+  }
+
+  > input + label {
+    -webkit-transition: all 500ms ease;
+    transition: all 500ms ease;    
+    cursor: pointer;    
+    display: inline-block;
+    -moz-user-select: -moz-none;
+    -webkit-user-select: none;
+    -ms-user-select: none;
+    user-select: none;
+    ${
+      ({props}) => `
+        font-size: ${ props?.styles?.fontSize || "14px" };  
+        border-radius: ${ props?.styles?.borderRadius || "5px" };  
+        background-color: ${ props?.styles?.backgroundColor || "#FFF" };  
+        border: ${ props?.styles?.border || "solid 2px #7d7d7d" };   
+        padding: ${ props?.styles?.padding || "0.1rem 0.3rem" };   
+        margin: ${ props?.styles?.margin || "5px" };   
+        color: ${ props?.styles?.defaultColor || "black" };   
+        width: ${ props?.styles?.width || "100px" };   
+        height: ${ props?.styles?.height || "auto" };   
+        text-align: ${ props?.styles?.textAlign || "center" }; 
+      `
+    }      
+  }
+
+  > input:checked + label {
+    -webkit-transition: all 500ms ease;
+    transition: all 500ms ease;   
+    ${
+      ({props}) => `
+        border-color: ${ props?.styles?.checkedColor || "#666699" };    
+        color: ${ props?.styles?.checkedColor || "white" };   
+        background-color: ${ props?.styles?.checkedBackgroundColor || "gray" };  
+      `
+    }     
+  }
+`;
+
+
+export const Chip = memo(
```

Comment:
whats the reason for using memo? do we see a signifanct delay whenever Chip is rerendered?

Reply:
Memo derives from memoization. It means that the result of the function wrapped in React.memo is saved in memory and returns the cached result if it's being called with the same input again.